### PR TITLE
Wip 1107 image CVE

### PR DESCRIPTION
**Pull request by nickmoreton**
This branch was started by Tom. He added the python-image lib to avoid using Pillow for student areas.

I tested it but it needed a couple of changes to check the file type with the different image library.

### CODE REVIEW

File: django-verdant/student_profiles/forms.py [hunk 1]
```diff
@@ -221,12 +222,16 @@ def clean_image(self):
...
+            "image/png",
+        )
+
         try:
-            dt = Image.open(img)
-            dt.load()
-            if dt.format.upper() not in ('PNG', 'JPEG', 'GIF', 'MPO'):
+            file_type = magic.from_buffer(img.read(), mime=True)
+            if not file_type in ALLOWED_FILE_TYPES:
                 raise forms.ValidationError(u'Only images of types JPEG and GIF are allowed. Please make sure that you save the image file as the specified format, instead of simply changing the file extension.')
```
Comment (richbrennan) [inline django-verdant/student_profiles/forms.py:234]: The error message says only JPEG and GIF are allowed, but PNG is also in the allowed file type list.

File: django-verdant/student_profiles/forms.py [hunk 2]
```diff
@@ -221,12 +222,16 @@ def clean_image(self):
...
+            "image/gif",
+            "image/png",
+        )
+
         try:
-            dt = Image.open(img)
-            dt.load()
-            if dt.format.upper() not in ('PNG', 'JPEG', 'GIF', 'MPO'):
+            file_type = magic.from_buffer(img.read(), mime=True)
+            if not file_type in ALLOWED_FILE_TYPES:
```
Comment (richbrennan) [inline django-verdant/student_profiles/forms.py:233]: Nitpick: `if file_type not in ALLOWED_FILE_TYPES:` reads a bit more nicely (is more Pythonic), from https://www.python.org/dev/peps/pep-0008/#programming-recommendations

File: django-verdant/student_profiles/forms.py [hunk 3]
```diff
@@ -221,12 +222,16 @@ def clean_image(self):
...
                 raise forms.ValidationError(u'Only images of types JPEG and GIF are allowed. Please make sure that you save the image file as the specified format, instead of simply changing the file extension.')
-
         except IOError:
             raise forms.ValidationError(u'Only images of types JPEG and GIF are allowed. Please make sure that you save the image file as the specified format, instead of simply changing the file extension.')
```
Comment (richbrennan) [inline django-verdant/student_profiles/forms.py:236]: Suggestion: Store the error message in a variable to avoid duplicating it.

**Review by richbrennan — COMMENTED** [review]
A few comments, thanks @nickmoreton !

File: django-verdant/student_profiles/forms.py (hunk 1 above)
Comment (nickmoreton) [reply django-verdant/student_profiles/forms.py:234]: Ah yes, thanks will change that just now.

File: django-verdant/student_profiles/forms.py (hunk 2 above)
Comment (nickmoreton) [reply django-verdant/student_profiles/forms.py:233]: Of course will change this too

File: django-verdant/student_profiles/forms.py (hunk 3 above)
Comment (nickmoreton) [reply django-verdant/student_profiles/forms.py:236]: :) Ditto

**Comment by nickmoreton** [comment]
Hey Rich. I have updated now with your suggestions, thank you.

One thing I now realise and it's probably no big deal but I think the error message in forms.py is probably not seen as they come from the JS file. Those messages are correct though.

**Review by richbrennan — APPROVED** [review]

**Pull request merged**